crates/cache/src/oci.rs

```diff
@@ -0,0 +1,130 @@
+//! Cache for OCI registry entities.
+
+use anyhow::{Context, Result};
+use oci_distribution::Reference;
+use tokio::fs;
+
+use std::path::{Path, PathBuf};
+
+const CONFIG_DIR: &str = "fermyon";
+const REGISTRY_CACHE_DIR: &str = "registry";
+const OCI_CACHE_DIR: &str = "oci";
+const MANIFESTS_DIR: &str = "manifests";
+const WASM_DIR: &str = "wasm";
+const DATA_DIR: &str = "data";
+
+/// Cache for registry entities.
+pub struct Cache {
+    /// Root directory for the cache instance.
+    pub root: PathBuf,
+}
+
+impl Cache {
+    /// Create a new cache given an optional root directory.
+    pub async fn new(root: Option<PathBuf>) -> Result<Self> {
+        let root = match root {
+            Some(root) => root,
+            None => dirs::config_dir()
```

Comment:
`config_dir` rather than, say, `data_local_dir` or `cache_dir`?

(I doubt we are too concerned about Windows at this stage but I'll mention that `config_dir` apparently roams on Windows... so...)

Reply:
This is using the same logic for getting the token as `deploy` and `login`.

Reply:
I would agree this should be `cache_dir`.

Reply:
Oh, is this for configuration settings?  I thought it was where the cache was actually stored?

Reply:
The intention is to use the same root directory for the configuration files _and_ the `registry` directory and cache.

Reply:
It might be useful to capture _why_ that is the intention - apologies if I seem obtuse but it's not jumping out at me.

Reply:
The main reason is having _one_ directory where Spin (and potentially future Fermyon projects) hold state by default.

My assumption so far has been that we would rather have _one_ default directory for this, compared to scattering config, state, and other things around on a user's system — such that if someone wants to remove Spin, they would just delete this directory.

I would be interested in hearing why we would consider splitting all this information in multiple directories _by default_.

Reply:
It's how it's usually done. For XDG for example, you expect anything a user might directly configure to be under `.config/` and you expect to be able to delete `.cache/` without any permanent problems.

Reply:
Ok — then it's a matter of changing the defaults. Happy to do that.

So is the global cache supposed to be placed in `dirs::cache_dir()` then, while everything else is in `dirs::config_dir()`?

Reply:
I have bad news I'm afraid, templates and plugins are already under `data_local_dir()`... though we can change that of course if we feel cache or config or something is better.

Reply:
Those are "data", arguably. These things are a little fuzzy...

Reply:
Yes, I agree - I'm just trying to figure out if we are trying to prioritise the "single base directory that you can delete in one command" over trying to fit things into XDG.

Reply:
If we are already splitting things into multiple directories, the registry work should follow that.

Reply:
This is now moving to the cache directory, and it will be side by side with templates.

If we decide on _where_ to move things, they should move together.